pkg/storage/secret/metadata/inline_store.go

```diff
@@ -0,0 +1,147 @@
+var _ contracts.InlineSecureValueSupport = (*inlineStorage)(nil)
+
+func ProvideInlineSecureValueSupport(
+	db contracts.SecureValueMetadataStorage,
+	access authlib.AccessClient,
+) (contracts.InlineSecureValueSupport, error) {
+	return &inlineStorage{
+		db:     db,
+		access: access,
+	}, nil
+}
+
+// TODO! this needs real attention :) and likely explicit support from the DB
+// inlineStorage is the actual implementation of the secure value (metadata) storage.
+type inlineStorage struct {
+	db     contracts.SecureValueMetadataStorage
+	access authlib.AccessChecker
+}
+
+// CanReference implements contracts.InlineSecureValueSupport.
+func (i *inlineStorage) CanReference(ctx context.Context, owner common.ObjectReference, values common.InlineSecureValues) (bool, error) {
+	actor, ok := authlib.AuthInfoFrom(ctx)
+	if !ok {
+		return false, apierrors.NewBadRequest("missing auth info")
+	}
+	if owner.Name == "" {
+		return false, apierrors.NewBadRequest("missing owner name")
+	}
+
+	// TODO? more efficient version of this based on database query?
+	// We want to check if the owner matches OR the identity in context have view permissions
+	for _, value := range values {
+		if !value.Create.IsZero() {
+			return false, apierrors.NewBadRequest("unable to create")
+		}
+		if value.Remove {
+			return false, apierrors.NewBadRequest("unable to remove")
+		}
+		if value.Name == "" {
+			return false, apierrors.NewBadRequest("expecting a name")
+		}
+
+		// ???? Read requires view?  do we have a read that does not require view
+		v, err := i.db.Read(ctx, xkube.Namespace(owner.Namespace), value.Name, contracts.ReadOpts{})
+		if err != nil || v == nil {
+			return false, apierrors.NewBadRequest("unable to reference secure value")
+		}
+
+		for _, ref := range v.OwnerReferences {
+			fmt.Printf("REF: %+v\n", ref)
+		}
+
+		// If the owner does not match... check if the request can read the name
+		ok, err := i.access.Check(ctx, actor, authlib.CheckRequest{
+			Verb:      utils.VerbGet,
+			Namespace: owner.Namespace,
+			Group:     secret.SchemeGroupVersion.Group,
+			Resource:  secret.SecureValuesResourceInfo.GroupResource().Resource,
+			Name:      value.Name,
+		})
+		if err != nil || !ok.Allowed {
+			return false, apierrors.NewBadRequest("not allowed to read value")
+		}
+	}
+
+	return true, nil
+}
+
+// UpdateSecureValues implements contracts.InlineSecureValueSupport.
+func (i *inlineStorage) UpdateSecureValues(ctx context.Context, owner common.ObjectReference, values common.InlineSecureValues) (common.InlineSecureValues, error) {
+	actor, ok := authlib.AuthInfoFrom(ctx)
+	if !ok {
+		return nil, apierrors.NewBadRequest("missing auth info")
+	}
+
+	keep := make(common.InlineSecureValues, len(values))
+	for key, secure := range values {
+		var prev *secret.SecureValue
+		if secure.Name != "" {
+			// The user may not be able to read the value if it is owned
+			prev, _ = i.db.Read(ctx, xkube.Namespace(owner.Namespace), secure.Name, contracts.ReadOpts{})
```

Comment:
So if the user cannot read the value, then we will set it to an empty string, and save it? What about in the case of the object being updated is the owner of that secret being referenced?

Reply:
TODO... this function is the real work :) the rest is plumbing